models/Asset.php

```diff
@@ -948,6 +947,14 @@ public function hasChildren()
         return false;
     }
 
+    /**
+     * @return Asset[]
+     */
+    public function getChildren()
+    {
+        return [];
+    }
+
```

Comment:
I suspect you added this method because the Interface requested it. But is it correct to always return an empty array? the Docs says it will return an array of asset objects (but that will never happen now).

What about throwing an error like "An asset cannot have children"?

Reply:
I add this, because the delete method use this method:
https://github.com/pimcore/pimcore/blob/8ae243d75f710701f62e7f4d1e97bb57fd767d38/models/Asset.php#L1016-L1021

The method is only in the Asset/Folder Class. I think we could add the method with an empty array. Why should this not compatible with the Docs? A folder can also have a empty array.

Reply:
`@return Asset[]` means an array of `Asset` objects, but this only returns an array, The Doc should say `@return []`

Reply:
The return type should be equal to all children classes. The Asset/Folder class returns a array with Assets. In the future we should add type hints, these could not changed with extend.

Reply:
If you load a list of assets (folder, images mixed) and call this function. The IDE should know it can only Assets in the array, If we add here array the IDE doesn't no it. That makes sense or not?

Reply:
But that code should be on the Listing, no?

I don't even know why this exists, it will never run:

```php
            // remove children
            if ($this->hasChildren()) {
                foreach ($this->getChildren() as $child) {
                    $child->delete(true);
                }
            }
```
because `hasChildren()` [always returns ](https://github.com/pimcore/pimcore/blob/8ae243d75f710701f62e7f4d1e97bb57fd767d38/models/Asset.php#L942-L948)`false` 

My IDE wants to declare it as `array` return type when I try it :)

![Skärmbild (254)](https://user-images.githubusercontent.com/279826/69860627-f0a59800-1296-11ea-9d40-be77eae3d508.png)

Reply:
The delete method is also valid for `\Pimcore\Asset\Folder`. There the hasChildren method can return true. Yes it would better to create a abstract class or a interface. Than the IDE wouldn't be show this notice.

Reply:
Oh, I thought `Folder` was `\Pimcore\Model\DataObject\Folder` - now I know there is *also* a `\Pimcore\Asset\Folder` :)

Now I need to understand why there are two different. In other words, keep it as is :)